### PR TITLE
Merge-Tree: Throw on invalid inserts

### PR DESCRIPTION
**Pull request by anthony-murphy**
In the past the Fluid Framework to the approach of avoiding throwing errors when errors could be ignored, however this has lead to issues where code consuming the Fluid Framework doesn't know it was doing something incorrectly, leading to compounding of errors which are hard to track as they are far away from the original problem site. This change cleans up error suppression on the merge tree insert path.

### CODE REVIEW

File: packages/dds/merge-tree/src/client.ts [hunk 1]
```diff
@@ -257,15 +257,13 @@ export class Client extends TypedEventEmitter<IClientEvents> {
 	 * @param pos - The position to insert the segment at
 	 * @param segment - The segment to insert
 	 */
-	public insertSegmentLocal(pos: number, segment: ISegment): IMergeTreeInsertMsg | undefined {
+	public insertSegmentLocal(pos: number, segment: ISegment): IMergeTreeInsertMsg {
 		if (segment.cachedLength <= 0) {
-			return undefined;
+			throw new UsageError("Cannot insert a 0 length segment.");
```
Comment (Abe27342) [inline packages/dds/merge-tree/src/client.ts:262]: we should add a changeset calling out this change and document expectation that `text` is non-empty on relevant `SharedString` methods. Wouldn't be surprised if users expected this to not throw.

Comment (anthony-murphy) [reply packages/dds/merge-tree/src/client.ts:262]: I'm actually debating if we should throw or not, as it did impact our tests. I wonder if it useful for callers to know, or if it is just annoying to have to check

Comment (anthony-murphy) [reply packages/dds/merge-tree/src/client.ts:262]: specifically, would a caller expect some side effect, that wasn't happening due to a zero length insert?

Comment (anthony-murphy) [reply packages/dds/merge-tree/src/client.ts:262]: The only benefit to our code is really the fuzz test change, as we never want to insert empty in fuzz due to it being a no-op

Comment (Abe27342) [reply packages/dds/merge-tree/src/client.ts:262]: IMO I mildly prefer the older behavior here, yeah.

File: packages/dds/sequence/src/test/fuzz/intervalRevertibles.fuzz.spec.ts [hunk 2]
```diff
@@ -34,8 +34,8 @@ import {
 	SharedStringFuzzFactory,
 	baseModel,
 	defaultFuzzOptions,
+	makeIntervalOperationGenerator,
 } from "./fuzzUtils.js";
-import { makeOperationGenerator } from "./intervalCollection.fuzz.spec.js";
```
Comment (anthony-murphy) [inline packages/dds/sequence/src/test/fuzz/intervalRevertibles.fuzz.spec.ts:38]: reusing from another test import was leading to all the test running again, so moved to fuzz utils

File: packages/dds/merge-tree/src/client.ts (hunk 1 above)
Comment (anthony-murphy) [reply packages/dds/merge-tree/src/client.ts]: i think i agree. i've rolled back that particular change, but kept some of the test, and less obvious failures.

**Review by Abe27342 — APPROVED** [review]

**Pull request merged**